### DISCUSSION TITLE
Better data-fetching library for nuxt

### DISCUSSION THREAD

**Discussion by Jarsen136** [kodadot/nft-gallery#10329] [Q&A] [answered]
Yup. Giving a key on `useAsyncData `  does not handle the cache automatically. To use the cache from nuxt, it should use `useNuxtData ` to manually manage the cache.

It's inconvenient to use tbh. I prefer some smart data-fetching library like `vue-query`, providing some powerful and convenient options like `staleTime`、 `queryKey ` 、`enabled ` etc.

ref: https://tanstack.com/query/latest/docs/framework/vue/guides/query-keys

_Originally posted by @Jarsen136 in https://github.com/kodadot/nft-gallery/pull/10299#discussion_r1606449003_
         
WDYT? Feel free to write down any other ideal option. @kodadot/internal-dev

**Comment by preschian** [comment]
> ref: [tanstack.com/query/latest/docs/framework/vue/guides/query-keys](https://tanstack.com/query/latest/docs/framework/vue/guides/query-keys)

I'm good with it. Ref: https://github.com/kodadot/nft-gallery/issues/4657#issuecomment-1378205624.

or,

for the profile case, cache it from the profile-workers side if possible

**Reply by Jarsen136** [reply]
> for the profile case, cache it from the profile-workers side if possible

It's not that proper because some addresses may edit their profile and also need to invalidate the cache. Caching from worker side may not be able to update flexiblely.

**Comment by Jarsen136** [comment, marked as answer]
Closed this discussion topic as I have solved it on #10299
